src/simulation/vizard/cielimInterface/zmqConnector.cpp

```diff
@@ -100,7 +101,17 @@ ImageData ZmqConnector::requestImage(size_t cameraId) {
     void* image = malloc(imageBufferLength*sizeof(char));
     memcpy(image, imagePoint, imageBufferLength*sizeof(char));
 
-    return  ImageData{imageBufferLength, image};
+    auto returnData = ImageData();
+    returnData.imageBuffer = image;
+    returnData.imageBufferLength = imageBufferLength;
+    returnData.centerOfBrightness = std::nullopt;
+
+    if (cobXMsgSize.value_or(0) > 0) {
```

Comment:
Should this just be `if (cobXMsgSize.has_value())` ? If there's no value the `returnData.centerOfBrightness` remains `std::nullopt` indicating there was no value.

Reply:
Done